### PR TITLE
Generate api docs for getters

### PR DESCRIPTION
**Pull request by devoncarew**
- generate api docs for getters (address part of https://github.com/dart-lang/web/issues/190)

Done as a few commits for easier review (the bulk of the work is in the 2nd commit).

This PR also switches to generating getters before setters for a getter/setter pair. The convention is to document the getter, and this lets the code read as `[docs]` `getter` `[setter]`.

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

**Review by kevmoo — APPROVED** [review]
Awesome!

File: tool/generator/translator.dart [hunk 1]
```diff
@@ -850,20 +882,28 @@ class Translator {
     };
   }
 
-  List<code.Method> _members(List<idl.Member> members) =>
-      [for (final member in members) ..._member(member)];
+  List<code.Method> _members(
+      List<idl.Member> members, MdnInterface? mdnInterface) {
+    return [
```
Comment (kevmoo) [inline tool/generator/translator.dart:887]: why no function syntax?

File: analysis_options.yaml [hunk 2]
```diff
@@ -35,7 +35,6 @@ linter:
   - no_runtimeType_toString
   - package_api_docs
   - prefer_const_declarations
-  - prefer_expression_function_bodies
```
Comment (kevmoo) [inline analysis_options.yaml:38]: Why? Is nice!

Comment (devoncarew) [reply analysis_options.yaml:38]: I like the ability to decide whether to use them or not based on which gives the best readability. This lint complains all the time, even when the non-expression version reads better.

File: tool/generator/translator.dart [hunk 3]
```diff
@@ -850,20 +882,28 @@ class Translator {
...
-  List<code.Method> _cssStyleDeclarationProperties() => [
-        for (final style in _cssStyleDeclarations)
-          ..._getterSetter(
-              fieldName: style,
-              getType: () => code.TypeReference((b) => b..symbol = 'String'),
-              isStatic: false,
-              readOnly: false),
-      ];
+  List<code.Method> _cssStyleDeclarationProperties() {
+    return [
```
[the remark this answers is not in the document]
Comment (devoncarew) [reply tool/generator/translator.dart:896]: I just found it more difficult to read and modify these methods as function expressions. I mostly stick with using them w/ trivial, ~1-line methods.

**Pull request merged**